keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt

```diff
@@ -227,12 +234,16 @@ class ApplicationLoadBalancerHandler(
                       ?: error("Keel does not support load balancers that are not in a VPC subnet"),
                     availabilityZones = lb.availabilityZones
                   ),
-                  internal = lb.scheme != null && lb.scheme!!.contains("internal", ignoreCase = true),
+                  internal = lb.scheme != null && lb.scheme!!.contains(
+                    "internal",
+                    ignoreCase = true
+                  ),
                   listeners = lb.listeners.map { l ->
                     ApplicationLoadBalancerSpec.Listener(
                       port = l.port,
                       protocol = l.protocol,
-                      certificateArn = l.certificates?.firstOrNull()?.certificateArn,
+                      certificate = l.certificates?.firstOrNull()
+                        ?.let { cloudDriverCache.certificateByArn(it.certificateArn).serverCertificateName },
```

Comment:
Use the cache to translate the ARN included in the River ALB response to the cert name